System/Posix/Time.hsc

```diff
@@ -39,3 +54,92 @@ epochTime = throwErrnoIfMinus1 "epochTime" (c_time nullPtr)
 
 foreign import ccall unsafe "__hsunix_time"
   c_time :: Ptr CTime -> IO CTime
+
+#ifdef HAVE_SETITIMER
+
+-- | Haskell expression of struct timeval: seconds and microseconds.
+data TimeVal = TimeVal {
+    tv_sec :: Int
```

Comment:
Does GHC still support any 32-bit CPUs with a 64-bit time_t?  Perhaps unofficially on some NetBSD system?  If so, `Int` may be large enough to correctly represent a C `time_t`.  The correct type here may be `CTime` from `Foreign.C.Types`